src/main/java/core/basesyntax/stategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.stategy;
+
+import core.basesyntax.model.Operation;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<Operation, FruitOperationHandler> fruitOperationHandlerMap;
+
+    public OperationStrategyImpl(Map<Operation,
+            FruitOperationHandler> fruitOperationHandlerMap) {
+        this.fruitOperationHandlerMap = fruitOperationHandlerMap;
+    }
+
+    @Override
+    public FruitOperationHandler getHandler(Operation operation) {
+        return fruitOperationHandlerMap.get(operation);
```

Comment:
Consider handling cases where the map does not contain a handler for the given operation. You might return a default handler or throw a more descriptive exception to indicate the absence of a handler.